src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java

```diff
@@ -174,15 +174,22 @@ protected void removeDoneFileQuietly(String rassInputFileName) {
                         + KFSConstants.BLANK_SPACE + objectDefinition.getObjectLabel() + " objects to process");
                 for (Object xmlObject : xmlObjects) {
                     T typedXmlObject = objectDefinition.getXmlObjectClass().cast(xmlObject);
-                    RassBusinessObjectUpdateResult<R> result = rassUpdateService.processObject(
-                            typedXmlObject, objectDefinition, documentTracker);
+                    RassBusinessObjectUpdateResult<R> result;
+                    try {
+                        result = rassUpdateService.processObject(typedXmlObject, objectDefinition, documentTracker);
+                    } catch (Throwable e) {
+                        LOG.error("updateBOs, caught an error while processing the objects", e);
+                        result = new RassBusinessObjectUpdateResult<>(objectDefinition.getBusinessObjectClass(), objectDefinition.printObjectLabelAndKeys(typedXmlObject), 
+                                RassObjectUpdateResultCode.ERROR, e.getMessage());
+                    }
```

Comment:
I realize that everything "should" be caught to set the error in the catch clause but in the event it does not, should "result" have a default value set outside the try-catch block?

Reply:
hmmm, good point, I will add an open else to the if block below that should handle the odd condition

Reply:
Upon looking further, I don't think there is a case where "result" wouldn't be set,  However, I did some refactoring that will hopefully make this cleaner.